aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -30,12 +30,15 @@ static inline void compare_base_kernel(Tensor& result, Tensor& indice,
   if (0 == result.numel()) {
     result.resize_(self_sizes);
   } else {
-    result = result.reshape(self_sizes);
+    //error out if result cannot be viewed as desired size
+    auto result_view = result.view(self_sizes);
+    result.set_(result_view);
```

Comment:
Nyehh, this isn't right, right? We should write into a contiguous buffer and then `copy_` it into the original source location. Otherwise if a user passes in a view, if you `set_` over it, the original memory being viewed won't get updated.

Reply:
The test is actually testing a view (slice), and memory behind it is updated, the command is something like
```
inp.max(dim=1, out=(full_indices[:,1], full_values[:,1]))
```

Reply:
why are we making up semantics on a per-function basis, though?  AFAIK we usually resize the output or throw an error if it's the wrong size.  But now we have a third choice, which is to re-view it?